### ISSUE TITLE
mIoU improves over validation set

### ISSUE THREAD

**Issue by manuel-88** [hellochick/ICNet-tensorflow#28]
Hey, I run your evaluation script and plotted the mIoU of every image over the whole validation dataset. The metric result is improving after every image. Why is it like this? In my opinion the result should alternate around some value. I saw that this mean IoU metric is updating the conf_matrix (sum up all pixel values) every step but this shouldn't affect the mean value?

Thanks for your help.

**Comment by hellochick** [comment]
Hey @manuel-88 , in the beginning of the evaluation process, not all classes in the confusion matrix have values, and this will affect the mIoU. Hence, you can discover that over a few steps, the mIoU will around the final value. Can this help you?

**Comment by manuel-88** [comment]
Ok, I understand now thank you. But this metric do not consider the frequency of pixels? means that classes with very less occurrence are same weighted than classes which occur often. 

Is this the metric which is generally used in semantic segmentation? My goal is it to compare the outcome of your framework with others and I want to be sure that the metric for all is the same.

**Comment by hellochick** [comment]
Hey @manuel-88 , sorry for my late reply. The issue you mentioned is a big problem, it really strange all classes have same weighted. However, this metric is generally used in semantic segmentation task. So you can compare with other works based on this metric.
